Source/WPF/MyMoney/Database/Money.cs

```diff
@@ -10632,67 +10579,30 @@ public bool Matches(QueryRow q)
             return false;
         }
 
-        public decimal GetCategorizedAmount(Category c)
-        {
-            if (c == null)
-            {
-                return this.amount;
-            }
 
-            if (this.IsDeleted || this.Status == TransactionStatus.Void)
-                return 0;
+        public decimal CurrencyNormalizedAmount(decimal Amount)

```

Comment:
The implementation of this needs to be fully optimized.  It will be used in tight loops over all transactions, so it needs to be faster.  We've moved from a simple property lookup (this.amount) to a complex method.

The `quickLookup` optimization in  FindCurrency could probably be optimized further by adding a singleton cache of previous currency, chances are we will see large swaths of transactions all in the same currency.

But most "loops" where this is needed could life out the "this.Account.Parent.Parent as MyMoney" thing and do it once.  And they could cache currency in the Account, so it could be just "t.amount * t.account.currency.ratio", could even cache the ratio in account and make it ""t.amount * t.account.currencyRatio".  That would make me happier.

Reply:
Done

Reply:
Looks good, I'd just remove [DataMember] from the new property.